File: Data Structures & Algorithms/anagram-groups/submission-0.go
func groupAnagrams(strs []string) [][]string {
    anagrams := make(map[[26]int][]string)

    for _ , str := range strs {
        count := [26]int{}
        for _ , ch := range str {
            count[ch - 'a']++
        }

        anagrams[count] = append(anagrams[count], str)

    }

    result := make([][]string, 0, len(anagrams))

    for _ , grp := range anagrams {
        result = append(result , grp)

    }

    return result
}

File: Data Structures & Algorithms/count-prefix-and-suffix-pairs-i/submission-0.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        count = 0

        for i in range(len(words)):
            for j in range(i+1 , len(words)):
                L = len(words[i])
                if words[i] == words[j][:L] and words[i] == words[j][-L:]:
                    count += 1
        

        return count